modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -301,7 +302,18 @@ def _get_new_qc(
         # reported dtypes from differing rows can be different based on the inference in
         # the limited data seen by each worker. We use pandas to compute the exact dtype
         # over the whole column for each column. The index is set below.
-        dtypes = cls.get_dtypes(dtypes_ids) if len(dtypes_ids) > 0 else None
+        dtypes, is_data_homogeneous = (
+            cls.get_dtypes_check_homogen(dtypes_ids)
+            if len(dtypes_ids) > 0
+            else (None, None)
+        )
+        if not is_data_homogeneous:
```

Comment:
Can we just move this check to `cls.get_dtypes` to handle all of the mismatched `dtype`s? We can throw the exception or warning there, right? That way we don't have to add a new method.

Reply:
Yes, let's use single `get_dtypes` for now. If it will be needed, separate function can be added in https://github.com/modin-project/modin/pull/3288.